### DISCUSSION TITLE
[A/L] (2021/22) Relazioni - Esercizio 2

### DISCUSSION THREAD

**Discussion by matypist** [sapienzastudentsnetwork/metodi-matematici-per-l-informatica#274] [Esercizi A-L]
![image](https://user-images.githubusercontent.com/69389461/152318228-4c04b13d-17cc-43aa-aa89-b735f77b9033.png)

**Comment by mildora00** [comment]
Non so se si può ragionare per esempi, ma io l'ho risolto così:

Ho assegnato ad R1 la relazione "è parente di" e ad R2 la proposizione "stessa età di".


Essendo aRb <=> a R1 b  e  a R2 b, R è:

- riflessiva <=> a "è parente di" a    e    a "stessa età di" a,  e sono vere entrambe.
- simmetrica <=>     a "è parente di" b   e  b"è parente di"a ;  a "stessa età di" b    e    b "stessa età di" a. Valgono entrambe.
- transitiva <=> a "è parente di" b   e  b"è parente di"a => a "è parente di" c
                                                                                   e
                          a "stessa età di" b    e    b "stessa età di" a =>  a "stessa età di" c , sono entambe valide.

Quindi R è anch'essa una relazione di equivalenza.

**Comment by matypist** [comment]
> Non so se si può ragionare per esempi, ma io l'ho risolto così:
> 
> Ho assegnato ad R1 la relazione "è parente di" e ad R2 la proposizione "stessa età di".
> 
> Essendo aRb <=> a R1 b e a R2 b, R è:
> 
>     * riflessiva <=> a "è parente di" a    e    a "stessa età di" a,  e sono vere entrambe.
> 
>     * simmetrica <=>     a "è parente di" b   e  b"è parente di"a ;  a "stessa età di" b    e    b "stessa età di" a. Valgono entrambe.
> 
>     * transitiva <=> a "è parente di" b   e  b"è parente di"a => a "è parente di" c
>       e
>       a "stessa età di" b    e    b "stessa età di" a =>  a "stessa età di" c , sono entambe valide.
> 
> 
> Quindi R è anch'essa una relazione di equivalenza.

@mildora00 occhio che l'esempio che hai adottato non è esattamente corretto rispetto la consegna dell'esercizio poiché **"è parente di"** (che dovrebbe rappresentare la relazione di equivalenza R1) **non è una relazione di equivalenza in quanto viola la proprietà transitiva**, infatti se Mario è parente di Luigi e Luigi è parente di Anna non è detto che Mario sia parente di Anna (Mario e Luigi potrebbero essere ad esempio cugini e avere dei parenti non in comune)

**Comment by matypist** [comment]
Inoltre mi si dice (cfr. @Exyss) che il ragionamento sia corretto ma che, come consiglio in vista dell'esame, in esercizi di questo tipo sia preferibile **dimostrare che non esista alcun caso in cui la condizione** (in questo caso "R è una relazione di equivalenza") **non sia verificata** anziché dimostrare un caso in cui è verificata

**Comment by mildora00** [comment]
> Inoltre mi si dice (cfr. @Exyss) che il ragionamento sia corretto ma che, come consiglio in vista dell'esame, in esercizi di questo tipo sia preferibile **dimostrare che non esista alcun caso in cui la condizione** (in questo caso "R è una relazione di equivalenza") **non sia verificata** anziché dimostrare un caso in cui è verificata



Va bene, grazie mille per i consigli. Gentilissimi entrambi.

**Comment by LuginiAndrea** [comment]
Ragioniamo sulle proprietè delle relazioni di equivalenza.
Dobbiamo dimostrare che
- $\forall a \in A \, aRa$
- $\forall a,b \in A \, aRb \Longrightarrow bRa$
-  $\forall a, b, c \in A \, aRb \wedge bRc \Longrightarrow aRc$

Queste proprietà sono rispettate da $R_1$ ed $R_2$
Verifichiamo se R rispetta queste proprietà:

- Riflessività: sappiamo che $aR_1a$ e $aR_2a$, ne consegue che $aRa$
- Simmetria: sappiamo che se $aR_1b$ allora $bR_1a$, e che se $aR_2b$ allora $bR_2a$, quindi $aR_1b \wedge aR_2b \Longrightarrow bR_1a \wedge bR_2a \Longrightarrow aRb \wedge bRa$
- Transitività: sappiamo che $aR_1b \wedge bR_1c \Longrightarrow aR_1c$, idem per $R_2$,  ne segue che
$\left(aR_1b \wedge aR_2b\right) \wedge \left(bR_1c \wedge bR_2c\right) \Longrightarrow aRb, bRc, \left(aR_1c \wedge aR_2c\right) \Longrightarrow aRb, bRc, aRc$

$R$ è una relazione d'equivalenza

**Comment by Exyss** [comment]
Per comodità di lettura, denotiamo $R$ come $\sim$, $R_1$ come $\sim_1$ e $R_2$ come $\sim_2$


### Riflessività :heavy_check_mark: 
$a \in A \implies a \sim_1 a, a \sim_2 a \implies a \sim a$

### Simmetria :heavy_check_mark: 
$a \sim b \iff a \sim_1 b \land a \sim_2 b \iff b \sim_1 a \land b \sim_2 a \iff b \sim a$

### Transitività :heavy_check_mark: 
$a \sim b \land b \sim c \iff a \sim_1 b \land b \sim_1 c  \land a \sim_2 b  \land b \sim_2 c \iff a \sim_1 c \land a \sim_2 c \iff a \sim c$
